server/src/sim.rs

```diff
@@ -109,6 +115,84 @@ impl Sim {
         Ok(())
     }
 
+    fn load_all_entities(&mut self, save: &save::Save) -> anyhow::Result<()> {
+        let mut read = save.read()?;
+        for node_hash in read.get_all_entity_node_ids()? {
+            let Some(entity_node) = read.get_entity_node(node_hash)? else {
+                continue;
+            };
+            let node_id = self.graph.from_hash(node_hash);
+            for entity_bytes in entity_node.entities {
+                let save_entity: SaveEntity = postcard::from_bytes(&entity_bytes)?;
+                self.load_entity(&mut read, node_id, save_entity)?;
+            }
+        }
+        Ok(())
+    }
+
+    fn load_entity(
+        &mut self,
+        read: &mut save::Reader,
+        node: NodeId,
+        save_entity: SaveEntity,
+    ) -> anyhow::Result<()> {
+        let entity_id = EntityId::from_bits(u64::from_le_bytes(save_entity.entity));
+        let mut entity_builder = EntityBuilder::new();
+        entity_builder.add(entity_id);
+        for (component_type, component_bytes) in save_entity.components {
+            if component_type == ComponentType::Position as u64 {
```

Comment:
nit: Use `ComponentType::try_from(component_type as i32)` then `match` on the result, and maybe abstract out a `load_component` helper for readability?

Reply:
Thanks! I hadn't realized that `ComponentType::try_from` was already written. I assume this is handled by `::prost::Enumeration`.